userspace/libsinsp/sinsp_filtercheck_fspath.cpp

```diff
@@ -349,7 +348,14 @@ uint8_t* sinsp_filter_check_fspath::extract(sinsp_evt* evt, OUT uint32_t* len, b
 			return NULL;
 		}
 
-		m_tstr = sinsp_utils::concatenate_paths(tinfo->get_cwd(), m_tstr);
+		if(m_tstr.front() != '/')
+		{
+			m_tstr = sinsp_utils::concatenate_paths(tinfo->get_cwd(), m_tstr.c_str());
```

Comment:
@federico-sysdig and @LucaGuerra hope you are ready for the next round :woman_facepalming:
Similar to the last PR we also have some issues here with the string. The possible trailing slash would not be removed otherwise. I have adjusted the unit tests to test this as well. Plus we also need to call `concatenate_paths` when the path is absolute. Absolute paths with path traversal happen on real-life servers.